### DISCUSSION TITLE
Run cell2location on NanoString Data

### DISCUSSION THREAD

**Discussion by Cirno-tensai9** [BayraktarLab/cell2location#91] [Q&A] [answered]
I recently tried the advanced workflow on NanoString data.
https://cell2location.readthedocs.io/en/latest/notebooks/cell2location_for_NanostringWTA.html

But it looks strange as 
(1)Training function as the tutorial said was deprecated.
(2)After applying "ignore_deprecation_exception=True". Training time becomes too long: more than 11 hours, while the worst GPU in the cluster I used is 1080Ti
(3) The input data format: it looks like a binary file which I don't know how to generate from the downloaded NanoString output (https://www.nanostring.com/products/geomx-digital-spatial-profiler/spatial-organ-atlas/).

Is there any latest functions for model training to use but haven't yet uploaded to the tutorial?
Can we get the script on how to transfer NanoString DSP data to the input format you supplied in the tutorial?

Best wishes,
Mengyuan

**Comment by vitkl** [comment]
Hi @Mengyuan-Li1996 

1. We are moving away from `pymc` which was very useful in prototyping this model to `pyro`+`scvi-tools` which provide tools for more advanced inference and scalability methods. We are progressively moving all models to the new interface. This warning is in place to prevent most users from applying the old version to their data. For now, NanoString model is supported only in pymc3. 

2. This is because default installation instructions do not setup `pymc` for use on GPU (analysis should take about as much as you are referencing on CPU). The way to address this is to install `pymc` and other dependencies as described in this targeted tutorial: https://cell2location.readthedocs.io/en/latest/installing_pymc3.html
@AlexanderAivazidis Could you please add a note that you need to follow a different installation procedure to the NanoString tutorial?

3. As a person in our group who developed the `cell2locations-WTA` model and works with this data, @AlexanderAivazidis can comment more on the format and how to convert their output to our inputs :)

Vitalii

**Reply by Cirno-tensai9** [reply]
Hi @vitkl ,

Thanks, I will find tutorial for pyro+scvi-tools and try it. 

Best wishes,
Mengyuan

**Comment by AlexanderAivazidis** [comment]
Hi @Mengyuan-Li1996 ,

could you share the names of the input files you have? Or send (a possibly downsampled version) to me? The data I worked with is from one of the first Nanostring-WTA runs and I think they changed the output files a few times since then. I will then see what I can do to make the input formats more clear and if possible make a script as you suggest. Generally three things are needed 1.) A matrix of unnormalized target probe counts, where 'target' means one gene or sometimes multiple genes with similar sequence (size: # ROIs x # Targets). This data should be in the 'TargetCountMatrix' in their output 2.) A matrix of unnormalized negative probe counts (size: # ROIs x # Negative Probes), those negative probes can usually be found in the 'ProbeCountMatrix' under a name such as 'NegProbe-WTA' (but they might have changed that since then). Optionally: 3.) Nuclei counts as a prior for cell numbers.

Best wishes,

Alexander

**Reply by Cirno-tensai9** [reply]
Hi @AlexanderAivazidis 

Sorry for the late reply, the WTA data I downloaded from https://www.nanostring.com/products/geomx-digital-spatial-profiler/spatial-organ-atlas/human-lymph-node/

All the two matrixes I didn't find a similar file in the downloading with filename grep, but there have a file Export1_InitialDataset.xlsx looks like can be used to extract these information.

But I still feel confused on how to use it on cell2loation.

Best wishes,
Mengyuan

**Comment by AlexanderAivazidis** [comment, marked as answer]
Hi Mengyuan,

this data doesn't quite look like the usual output from the Nanostring pipelines. Also it looks like they don't provide the Negative Probe Counts. So the best way would be to get the 'TargetCountMatrix' from the file called Export3_BiologicalProbeQC and supply that to the standard cell2locatin pipeline. Results should still be acceptable like that. Otherwise, you can also contact the owners of the dataset and ask for the 'ProbeCountsMatrix', which should contain negative probe counts under names, such as 'NegProbe-WTA'. 

Best wishes,

Alexander
